config/boards/orangepi5-plus.conf

```diff
@@ -28,9 +30,39 @@ function post_family_tweaks__orangepi5plus_naming_audios() {
 	return 0
 }
 
-# Override uboot family config for this board; let's avoid conditionals in family config.
-function post_family_config__orangepi5plus_use_vendor_uboot() {
+# Use vendor-provided U-Boot for 'legacy' kernel branch; let's avoid conditionals in family config.
+function post_family_config_branch_legacy__orangepi5plus_use_vendor_uboot() {
 	BOOTSOURCE='https://github.com/orangepi-xunlong/u-boot-orangepi.git'
 	BOOTBRANCH='branch:v2017.09-rk3588'
 	BOOTPATCHDIR="legacy/u-boot-orangepi5-rk3588"
 }
+
+# Use vendor-provided U-Boot for 'vendor' kernel branch; let's avoid conditionals in family config.
+function post_family_config_branch_vendor__orangepi5plus_use_vendor_uboot() {
+	BOOTSOURCE='https://github.com/orangepi-xunlong/u-boot-orangepi.git'
+	BOOTBRANCH='branch:v2017.09-rk3588'
+	BOOTPATCHDIR="legacy/u-boot-orangepi5-rk3588"
+}
+
+# Mainline U-Boot for edge kernel
+function post_family_config_branch_edge__nanopct6_use_mainline_uboot() {
```

Comment:
can you rename nanopct6 -> orangepi5plus. otherwise the pr is ok and tested with spi+nvme boot